qiskit/algorithms/linear_solvers/matrices/tridiagonal.py

```diff
@@ -0,0 +1,210 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Hamiltonian simulation of tridiagonal Toeplitz symmetric matrices."""
+
+from typing import Optional
+
+import numpy as np
+from scipy.sparse import diags
+
+from qiskit.circuit import QuantumCircuit, QuantumRegister, AncillaRegister, Qubit
+
+
+class Tridiagonal(QuantumCircuit):
+    """Class of tridiagonal Toeplitz symmetric matrices"""
+
+    def __init__(self, num_state_qubits: int, main_diag: float, off_diag: float,
+                 tolerance: Optional[float] = None, time: Optional[float] = None,
+                 trotter: Optional[int] = 1, name: str = 'tridi') -> None:
+
+        qr_state = QuantumRegister(num_state_qubits)
+        if num_state_qubits > 1:
+            qr_ancilla = AncillaRegister(max(1, num_state_qubits - 1))
+            super().__init__(qr_state, qr_ancilla, name=name)
+        else:
+            super().__init__(qr_state, name=name)
+
+        self._num_state_qubits = None
+
+        self._main_diag = main_diag
+        self._off_diag = off_diag
+        self._tolerance = tolerance if tolerance is not None else 1e-2
+        self._time = time if time is not None else 1
```

Comment:
Why make `time` Optional and then detect None here and change that to 1. Would it not be better to simply have `time: float = 1.0` as the parameter?

Also is '`time` a good name here - I imagine this might be `time_limit` where its an upper bound on the amount of time allowed before returning a result but presumably it can return sooner. Docstring for the args, as mentioned above would presumable state what this is - and the units - I guess its time in seconds.